src/CalcManager/CEngine/scidisp.cpp

```diff
@@ -215,10 +215,11 @@ int CCalcEngine::IsNumberInvalid(const wstring& numberString, int iMaxExp, int i
 \****************************************************************************/
 vector<uint32_t> CCalcEngine::DigitGroupingStringToGroupingVector(wstring_view groupingString)
 {
-    vector<uint32_t> grouping{};
+    vector<uint32_t> grouping;
     uint32_t currentGroup = 0;
     wchar_t* next = nullptr;
-    for (const wchar_t* itr = groupingString.data(); *itr != L'\0'; ++itr)
+    const wchar_t* end = groupingString.data() + groupingString.size();
+    for (auto itr = groupingString.data(); itr != end; ++itr)
```

Comment:
Should we use     
` for (auto it = groupingString.cbegin(); it != groupingString.cend(); ++it)` 

instead?

Reply:
That was my initial thought but the way itr was being used inside the loop as a const wchar_t* made that challenging. Maybe the iterator could have been casted but it felt weird.

Reply:
That was my initial thought but the way itr was being used inside the loop as a const wchar_t* made that challenging. Maybe the iterator could have been casted but it felt weird.

Reply:
Got it! We can keep it like that, but I would suggest to only use size() or length() but not both in the same function (line 236) and call `groupingString.data()` only one time if possible